packages/wonder-blocks-button/components/clickable-behavior.js

```diff
@@ -0,0 +1,190 @@
+// @flow
+/**
+ * Add hover, focus, and active status updates to a clickable component.
+ *
+ * Via mouse:
+ * 1. Hover over button -> hover state
+ * 2. Mouse down -> active state
+ * 3. Mouse up -> default state
+ * 4. Press tab -> focus state
+ *
+ * Via touch:
+ * 1. Touch down -> press state
+ * 2. Touch up -> default state
+ *
+ * Via keyboard:
+ * 1. Tab to focus -> focus state
+ * 2. Keydown (spacebar/enter) -> active state
+ * 3. Keyup (spacebar/enter) -> focus state
+ *
+ * ClickableBehavior accepts a function as 'children' which is passed state and
+ * an object containing event handlers. The 'children' function should return
+ * a clickable React Element of some sort.
+ *
+ * Example:
+ *
+ * class MyClickableComponent extends React.Component<Props> {
+ *     render() {
+ *         return <ClickableBehavior
+ *             disabled={this.props.disabled}
+ *             onClick={this.onClick}
+ *         >
+ *             {({hovered}, handlers) =>
+ *                 <RoundRect
+ *                      textcolor='white'
+ *                      backgroundColor={hovered ? 'red' : 'blue'}/>}
+ *                      {...handlers}
+ *                 >
+ *                      {this.props.children}
+ *                 </RoundRect>
+ *             }
+ *         </ClickableBehavior>
+ *     }
+ * }
+ *
+ * This follows a pattern call 'Function as Child Components':
+ * https://medium.com/merrickchristensen/function-as-child-components-5f3920a9ace9
+ */
+import React from "react";
+
+type Props = {
+    children: (state: State, handlers: Handlers) => React$Element<*>,
```

Comment:
I like that this is general enough that if people do need to make other things clickable in a cross platform way they can.

Reply:
This makes me wonder if we should make this component public (exposed in the docs and through wonder-blocks-button) so that people can use it in their code?